clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ConnectionState.java

```diff
@@ -139,6 +139,12 @@ public void initClusterConnection(Executor asyncWorker) {
   private void reconnect() {
     while (true) {
       try {
+        try {
+          //Ensure full closure of existing connection
+          clusterConnection.close();
+        } catch (IOException | ConnectionClosedException | IllegalStateException e) {
+          LOGGER.debug("Exception closing previous cluster connection", e);
```

Comment:
Would this be better at INFO or WARN since it could have an impact the subsequent `connect` call?  Or are you expecting this to be an irrelevant issue often enough to want to suppress the noise?

Reply:
The `ConnectionShutdownException` might the the most frequent noise .... you could eat this one and log anything else.